docs/messaging/messaging-spans.md

```diff
@@ -153,36 +153,35 @@ in such a way that it cannot be changed by intermediaries.
 
 ### Span name
 
-The span name SHOULD be set to the message destination name and the name of the operation being performed (as captured in [`messaging.operation.name`](../attributes-registry/messaging.md)) in the following format:
+Messaging spans SHOULD follow the overall [guidelines for span names](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/trace/api.md#span).
 
-```
-<destination name> <operation name>
-```
+<!-- markdown-link-check-disable -->
+<!-- HTML anchors are not supported https://github.com/tcort/markdown-link-check/issues/225-->
+The **span name** SHOULD be `{messaging.operation.name} {destination}` (see below for the exact definition of the [`{destination}`](#destination-placeholder) placeholder).
+<!-- markdown-link-check-enable -->
 
-If the operation name is not specified by the messaging system, then the operation type as defined in [Operation types](#operation-types) SHOULD be used:
+If neither `{messaging.operation.name}` nor `{destination}` are available, span name SHOULD be `{messaging.system}`.
 
-```
-<destination name> <operation type>
-```
+Semantic conventions for individual messaging systems MAY specify different span name format and then MUST document it in [semantic conventions for specific messaging technologies](#semantic-conventions-for-specific-messaging-technologies).
 
-The destination name SHOULD only be used for the span name if it is known to be of low cardinality (cf. [general span name guidelines](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.33.0/specification/trace/api.md#span)).
-This can be assumed if it is statically derived from application code or configuration.
-Wherever possible, the real destination names after resolving logical or aliased names SHOULD be used.
-If the destination name is dynamic, such as a [conversation ID](#conversations) or a value obtained from a `Reply-To` header, it SHOULD NOT be used for the span name.
-In these cases, an artificial destination name that best expresses the destination, or a generic, static fallback like `"(anonymous)"` for [anonymous destinations](#temporary-and-anonymous-destinations) SHOULD be used instead.
+The <span id="destination-placeholder">`{destination}`</span> SHOULD describe the entity that the operation is performed against
+and SHOULD adhere to one of the following values, provided they are accessible:
```

Comment:
```suggestion
and SHOULD adhere to the first applicable option from the following list:
```

Maybe that makes it a bit clearer?

Reply:
We discovered fallback to be problematic in DB semconv.

Let's say someone publishes a message to a temp destination and there is no template.
This operation usually (on other destinations) is called `publish {detaintion.name}`, but since it's a temp operation, the fallback would result in `publish {address:port}`.
So fallback would result in inconsistent operation names.

What we want is the entity operation is done against (broker or a specific queue|topic).
So instead of falling back, we'd rather not populate destination part of the span name.